sql/hive/src/main/scala/org/apache/spark/sql/hive/TableReader.scala

```diff
@@ -248,11 +248,16 @@ class HadoopTableReader(
         // SPARK-13709: For SerDes like AvroSerDe, some essential information (e.g. Avro schema
         // information) may be defined in table properties. Here we should merge table properties
         // and partition properties before initializing the deserializer. Note that partition
-        // properties take a higher priority here. For example, a partition may have a different
-        // SerDe as the one defined in table properties.
+        // properties take a higher priority here except for the Avro table properties
+        // to support schema evolution: in that case the properties given at table level will
+        // be used (for details please check SPARK-26836).
+        // For example, a partition may have a different SerDe as the one defined in table
+        // properties.
         val props = new Properties(tableProperties)
-        partProps.asScala.foreach {
-          case (key, value) => props.setProperty(key, value)
+        partProps.asScala.filterNot { case (k, _) =>
+          k == AvroTableProperties.SCHEMA_LITERAL.getPropName() && tableProperties.containsKey(k)
```

Comment:
`&& tableProperties.containsKey(k)` looks risky to me and beyond this PR's test coverage. According to the test case, it looks like `k == AvroTableProperties.SCHEMA_LITERAL.getPropName()` is enough to pass the test case. Did I understand correctly?